akka-persistence-typed/src/main/scala/akka/persistence/typed/Retention.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed
+
+/** Retention bridges snapshot and journal behavioral flow.
+  * Can describe the algorithms of retention used by snapshot
+  * and journal logic, where applicable, focusing on that
+  * which is shared.
+  */
+
+/**
+  * Retention criteria.
+  *
+  * @param snapshotSize        Snapshots are used to reduce playback/recovery times. This defines
+  *                            when a new snapshot is persisted. Defaults to a count algorithm,
+  *                            with 1000 messages, such that on 1001, the next snapshot is taken,
+  *                            and if successful, the events comprising the snapshot
+  *                            are deleted to free up space. Defaults to 1000.
+  *
+  * @param keepNSnapshots      After a snapshot is successfully completed,
+  *                             - if 2: retain last maximum 2 *`snapshot-size` events and 3 snapshots
+  *                               (2 old + latest snapshot)
+  *                             - if 0: all events with equal or lower sequence number will not be retained.
+  *                            Defaults to a count of 2 snapshots.
+  *
+  * @param retainSnapshotsOnly Opt-in ability to not delete events, only delete snapshots.
```

Comment:
shouldn't this be named `deleteSnapshotsOnly`? doesn't retain mean "keep"

Reply:
Works for me. We retain N-messages and N-Snapshots and delete what is in the criteria for the logic so to me this is about logic on what to retain and basically TTL the rest.

Reply:
IMO to decide what to delete, I'd be basing it on what I want to retain at any given time, based on expectation of size etc.

Reply:
When I read retainSnapshotsOnly=true I think it means that events will be deleted and snapshots will kept.

retain=continue to have (something); keep possession of.

but I think you mean the opposite

Reply:
The issue states, `a way to NOT delete events at all but only delete snapshots`
so I was confused and thought always delete events on save snapshot success, opt-in to not delete events. So to clarify:
```scala
deleteMessagesOnSnapshot=true` - events will be deleted if true, defaults to false
```